app/score_calculator/yaku_check/block_yaku/two_terminal_chows.py

```diff
@@ -0,0 +1,22 @@
+from typing import Final
+
+from app.score_calculator.block.block import Block
+from app.score_calculator.enums.enums import BlockType
+
+SIZE: Final[int] = 2
+TERMINAL_GAP: Final[int] = 6
+
+
+def valid_check_two_terminal_chows(blocks: list[Block]) -> bool:
+    if len(blocks) != SIZE:
+        return False
+    return all(block.type == BlockType.SEQUENCE for block in blocks)
+
+
+def check_two_terminal_chows(blocks: list[Block]) -> bool:
+    if not valid_check_two_terminal_chows(blocks):
+        return False
+    return (
+        len({block.tile.get_type() for block in blocks}) == 1

```

Comment:
이렇게하면 중간 저장변수를 내부적으로 생성하며, short circuit eval을 하지 못합니다.
다음과 같은 방식들이 일반적인 practice 입니다.
```suggestion
        from itertools import groupby
        group = groupby(block.tile.get_type() for block in blocks)
        return next(group, True) and not next(group, False)
```

or

```suggestion
        return all(blocks[0] == block for block in blocks)
```
위의 방식이 아주 약간 빠른것으로 알려져있으나, 보통 가독성 때문에 아래 코드를 사용합니다.

Reply:
group by 를 이럴때사용하는군요 bb